mypy.ini

```diff
@@ -0,0 +1,107 @@
+[mypy]
+check_untyped_defs = True
+strict_optional = True
+[mypy-docs.macros]
+ignore_errors = True
+[mypy-up42.asset]
+ignore_errors = False
+[mypy-up42.asset-searcher]
+ignore_errors = False
+[mypy-up42.auth]
+ignore_errors = True
+[mypy-up42.catalog]
+ignore_errors = False
+[mypy-up42.estimation]
+ignore_errors = True
+[mypy-up42.initialization]
+ignore_errors = True
+[mypy-up42.job]
+ignore_errors = True
+[mypy-up42.jobcollection]
+ignore_errors = True
+[mypy-up42.jobtask]
+ignore_errors = True
+[mypy-up42.main]
+ignore_errors = True
+[mypy-up42.order]
+ignore_errors = True
+[mypy-up42.project]
+ignore_errors = True
+[mypy-up42.stac_client]
+ignore_errors = True
+[mypy-up42.storage]
+ignore_errors = True
+[mypy-up42.tasking]
+ignore_errors = True
+[mypy-up42.tools]
+ignore_errors = True
+[mypy-up42.utils]
+ignore_errors = True
+[mypy-up42.viztools]
+ignore_errors = True
+[mypy-up42.webhooks]
+ignore_errors = True
+[mypy-up42.workflow]
+ignore_errors = True
+[mypy-tests.conftest]
+ignore_errors = True
+[mypy-tests.context]
+ignore_errors = True
+[mypy-tests.test_asset]
+ignore_errors = True
+[mypy-tests.test_auth]
+ignore_errors = True
+[mypy-tests.test_catalog]
+ignore_errors = True
```

Comment:
The same as above. I'd recommend to set catalog.py ignore_errors=True and focus on asset .py and test_asset.py

Reply:
I see.
For this PR, let's go ahead with both: asset and catalog and their corresponding tests.